src/components/DatePicker/index.tsx

```diff
@@ -1,36 +1,132 @@
-import React, { useCallback } from 'react';
+import type React from 'react';
+import { useCallback, useState } from 'react';
 
 import { useTheme } from '@mui/material';
+import { LocalizationProvider, type DateView } from '@mui/x-date-pickers';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import dayjs, { type Dayjs } from 'dayjs';
 
-import { TextInput } from '../TextInput';
+import { Flex } from '../Flex';
+import { Icon } from '../Icon';
+import { Typography } from '../Typography';
 
-export const DatePicker = ({
-  id,
-  label,
-  value,
-  onChange,
-}: {
-  id?: string;
+import {
+  ChevronLeft,
+  ChevronRight,
+  ChevronDown,
+  FlexToggle,
+  StyledDatePicker,
+  StyledPaper,
+  FlexCalendarFooter,
+} from './styles';
+
+type DatePickerProps = {
   label: string;
-  value: string;
-  onChange: (value: string) => void;
-}) => {
+  initialDate?: string;
+  onChange: (date: string) => void;
+};
+
+export const DatePicker: React.FC<DatePickerProps> = ({ label, initialDate, onChange }) => {
   const theme = useTheme();
-  const onDateChange = useCallback(
-    (value: string) => {
-      onChange(value);
-    },
-    [onChange]
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [currentView, setCurrentView] = useState<DateView>('day');
+  const [selectedDate, setSelectedDate] = useState<Dayjs>(dayjs(initialDate));
+
+  const goToPreviousMonth = useCallback(() => setSelectedDate(curr => curr.subtract(1, 'month').set('date', 1)), []);
+
+  const goToNextMonth = useCallback(() => setSelectedDate(curr => curr.add(1, 'month').set('date', 1)), []);
+
+  const goToPreviousYear = useCallback(() => setSelectedDate(curr => curr.subtract(1, 'year').set('date', 1)), []);
+
+  const goToNextYear = useCallback(() => setSelectedDate(curr => curr.add(1, 'year').set('date', 1)), []);
+
+  const toggleMonthView = useCallback(() => {
+    setCurrentView(currView => (currView === 'month' ? 'day' : 'month'));
+  }, []);
+
+  const toggleYearView = useCallback(() => {
+    setCurrentView(currView => (currView === 'year' ? 'day' : 'year'));
+  }, []);
+
+  const toggleOpen = useCallback(() => {
+    setIsOpen(curr => !curr);
+  }, []);
+
+  const renderOpenPickerIcon = () => (
```

Comment:
pls move all the `render` functions out of this component... its an anti pattern to construct react trees as variables inside renders like this